File: qf-auto-repair-pojo/src/main/java/com/qianfeng/entity/RepairOrder.java
package com.qianfeng.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@TableName("repair_orders")
@Data
public class RepairOrder {
    public static final String TO_BE_REPAIR = "待维修";
    public static final String UNDER_REPAIR = "维修中";
    public static final String COMPLETED = "已完成";
    public static final String CANCELLED = "已取消";
    @TableId(type = IdType.AUTO)
    private Integer orderId;

    @TableField("vehicle_id")
    private Integer vehicleId;

    @TableField("customer_id")
    private Integer customerId;

    @TableField(value = "order_date",fill = FieldFill.INSERT)
    private LocalDateTime orderDate;

    @TableField("status")
    private String status;

    //orderCost 属性
    @TableField("order_cost")
    private BigDecimal orderCost;

    @TableField("technician_id")
    private Integer technicianId; // 对应着employee表中的employeeId
}

File: src/types/repairOrder.ts
export type Status = "待维修" | "维修中" | "已完成" | "已取消";

export interface RepairOrderQueryDTO {
  orderId?: number;
  vehicleId?: number;
  customerId?: number;
  orderDate?: string; // 对应LocalDateTime
  status?: Status;
  orderCost?: number;
  technicianId?: number;
  vehicleLicensePlate?: string;
  current: number; // 分页参数
  size: number; // 分页参数
}

export interface RepairOrderDTO {
  orderId?: number;
  vehicleId?: number;
  customerId?: number;
  orderDate?: string; // 创建时可不传，由后端生成
  status?: Status;
  vehicleLicensePlate?: string;
  orderCost?: number;
  technicianId?: number;
}

export interface RepairOrderVO {
  orderId: number;
  vehicleId: number;
  customerId: number;
  orderDate: number[];
  status: Status;
  orderCost: number;
  technicianId?: number;
  vehicleLicensePlate: string;
  customerName: string;
  phone: string;
  technicianName?: string;
}
